packages/provider/waallet/background/provider.ts

```diff
@@ -82,13 +82,16 @@ export class WaalletBackgroundProvider {
       // TODO: When `to` is empty, it should estimate gas for contract creation
       return
     }
-    if (tx.from && tx.from !== (await this.account.getAddress())) {
+    if (
+      tx.from &&
+      ethers.getAddress(tx.from) !== (await this.account.getAddress())
+    ) {
       throw new Error("Address `from` doesn't match connected account")
     }
     // TODO: Use account's entry point
     const [entryPointAddress] = await this.bundler.getSupportedEntryPoints()
     const userOpCall = await this.account.createUserOperationCall({
-      to: tx.to,
+      to: ethers.getAddress(tx.to),
```

Comment:
Q: Is it required the conversion here?

Reply:
In longer view it would be better to have a class to denote the address value, just like #49. We can first think more about the question: is there any good place to do the conversion for us?

Reply:
OK! I just reviewed the content of #49.
It introduces a `UserOperation` class for seamless handling and offers a `data()` method, mainly converting `BigInt` to `Hex`.
Using classes helps centralize data processing. If conversion methods change in the future, modifying the class suffices for global adjustments.
For instance, we can add `EthEstimateGas` and `EthSendTransaction` classes for implicit address normalization.